routes/validators.go

```diff
@@ -354,6 +356,127 @@ func (fes *APIServer) GetValidatorByPublicKeyBase58Check(ww http.ResponseWriter,
 	}
 }
 
+// GetCurrentEpochProgressResponse encodes the current epoch entry, the leader schedule for it, and the
+// progress throughout the epoch. Based on the data returned, the client can determine the chain's
+// progress through the epoch, the current leader and all upcoming leaders.
+type GetEpochProgressResponse struct {
+	// The full epoch entry object
+	EpochEntry     lib.EpochEntry  `safeForLogging:"true"`
```

Comment:
Normally I'd say we need to create a new response type, but since the epoch entry is just a bunch of uint64s and int64s, this is chill.